commit 6c27b5230e356ce2f4e259bef17a78864510d7ba
Author: Hannes Reinecke <hare@suse.de>
Date:   Tue Jan 8 12:46:58 2019 +0100

    nvme-multipath: zero out ANA log buffer
    
    [ Upstream commit c7055fd15ff46d92eb0dd1c16a4fe010d58224c8 ]
    
    When nvme_init_identify() fails the ANA log buffer is deallocated
    but _not_ set to NULL. This can cause double free oops when this
    controller is deleted without ever being reconnected.
    
    Signed-off-by: Hannes Reinecke <hare@suse.com>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/nvme/host/multipath.c b/drivers/nvme/host/multipath.c
index c27af277e14e..815509dbed84 100644
--- a/drivers/nvme/host/multipath.c
+++ b/drivers/nvme/host/multipath.c
@@ -556,6 +556,7 @@ int nvme_mpath_init(struct nvme_ctrl *ctrl, struct nvme_id_ctrl *id)
 	return 0;
 out_free_ana_log_buf:
 	kfree(ctrl->ana_log_buf);
+	ctrl->ana_log_buf = NULL;
 out:
 	return error;
 }
@@ -563,5 +564,6 @@ int nvme_mpath_init(struct nvme_ctrl *ctrl, struct nvme_id_ctrl *id)
 void nvme_mpath_uninit(struct nvme_ctrl *ctrl)
 {
 	kfree(ctrl->ana_log_buf);
+	ctrl->ana_log_buf = NULL;
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1390_6c27b5230e356ce2f4e259bef17a78864510d7ba.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1390_6c27b5230e356ce2f4e259bef17a78864510d7ba.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
